packages/ics721/src/types.rs

```diff
@@ -0,0 +1,68 @@
+use cosmwasm_schema::{cw_serde, schemars::JsonSchema};
+use cosmwasm_std::Binary;
+use serde::{Deserialize, Serialize};
+
+use crate::{ibc::NonFungibleTokenPacketData, token_types::ClassId};
+
+#[derive(Serialize, Deserialize, JsonSchema, Clone, Debug, PartialEq)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[schemars(crate = "cosmwasm_schema::schemars")]
+#[serde(crate = "cosmwasm_schema::serde")]
+pub struct Ics721Memo {
+    pub callbacks: Option<Ics721Callbacks>,
+}
+
+/// The format we expect for the memo field on a send
+#[cw_serde]
+pub struct Ics721Callbacks {
+    /// Data to pass with a callback on source side (status update)
+    /// Note - If this field is empty, no callback will be sent
+    pub src_callback_msg: Option<Binary>,
+    /// The address that will receive the callback message
+    pub src_msg_receiver: Option<String>,
+    /// Data to pass with a callback on the destination side (ReceiveNftIcs721)
+    /// Note - If this field is empty, no callback will be sent
+    pub dest_callback_msg: Option<Binary>,
+    /// The address that will receive the callback message
+    pub dest_msg_receiver: Option<String>,
+}
+
+/// A message is that is being called on receiving the NFT after transfer was completed.
+/// Receiving this message means that the NFT was successfully transferred.
+/// You must verify this message was called by an approved ICS721 contract, either by code_id or address.
+#[cw_serde]
+pub struct Ics721ReceiveMsg {
+    pub original_packet: NonFungibleTokenPacketData,
```

Comment:
I'd just name it `packet` or `nft_data`.

Reply:
I would like to keep the original here, to make it obvious that this is the original packet that was sent, and not anything else.